### ISSUE TITLE
Help Wanted: subclass of Reader that uses filestore

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/bluesky#560]
@CJ-Wright

```python
class ReaderWithFileStore(Reader):

    def __init__(self, *args, fs, **kwargs):  # fs is a required keyword arg
        """
        ... copy docstring from Reader...
        fs : FileStore
            FileStore object that supports inserting resource and datum documents
        """
        super().__init__(*args, **kwargs)
        self.fs = fs
        self._resource_id = None

    def stage(self):
        self._resource_id = self.fs.insert_resource(...)

    def read(self):
        result = {}
        for name, val in super().read().items():
            # save val (probably an array) to a file somewhere, like ~/.cache/fs_cache
            # make a datum_id and figure out datum_kwargs
            self.fs.insert_datum(...)
            result[name] = datum_id
        return result

    def unstage(self):
        self._resource_id = None
```

**Comment by danielballan** [comment]
@CJ-Wright, Tom has some related work going on here, but it doesn't conflict with this. Carry on.

**Comment by CJ-Wright** [comment]
@danielballan Don't we need to call insert_resource for each file created? Each has a new path for extraction.

**Comment by danielballan** [comment]
Closed by #574. Thanks @CJ-Wright!
